kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/ErrorTransform.java

```diff
@@ -0,0 +1,380 @@
+ * this SMT
+ *
+ * <p>Actual value converter, key converter, and header converter are configured on the
+ * `transforms.xxx` key where xxx is the name of this transform. See example for how properties are
+ * passed and namespaced
+ *
+ * <p>"transforms": "tab", "transforms.tab.type":
+ * "io.tabular.iceberg.connect.transform.managed.ManagedTransform",
+ * "transforms.tab.value.converter": "org.apache.kafka.connect.storage.StringConverter",
+ * "transforms.tab.value.converter.some_property: "...", "transforms.tab.key.converter":
+ * "org.apache.kafka.connect.storage.StringConverter", "transforms.tab.key.converter.some_property":
+ * "...", "transforms.tab.smts" : "some.java.class,some.other.java.class",
+ * "transforms.tab.smts.prop1" : "some_property_for_the_smts"
+ *
+ * <p>This should not be used with any other SMT. All SMTs should be added to "transforms.tab.smts".
+ *
+ * <p>It returns a special Map of String -> Object "original" : Map of String -> Object containing
+ * the key,value, and header bytes of the original message "transformed" : [null, Struct, Map, etc.]
+ * of whatever the deserialized record is (after transformation if SMTs are configured)
+ *
+ * <p>The original payload can be used in the Iceberg Connector if the record cannot be transformed
+ * to an Iceberg record so that the original kafka message can be stored in Iceberg at that point.
+ *
+ * <p>If any of the key, value, header deserializers or SMTs throw an exception a failed record is
+ * constructed that contains kafka metadata, exception/location information, and the original
+ * key/value/header bytes.
+ */
+public class ErrorTransform implements Transformation<SinkRecord> {
+
+  public static class TransformInitializationException extends RuntimeException {
+    TransformInitializationException(String errorMessage) {
+      super(errorMessage);
+    }
+
+    TransformInitializationException(String errorMessage, Throwable err) {
+      super(errorMessage, err);
+    }
+  }
+
+  public static class PropsParser {
+    static Map<String, ?> apply(Map<String, ?> props, String target) {
+      return props.entrySet().stream()
+          .filter(
+              entry ->
+                  (!Objects.equals(entry.getKey(), target)) && (entry.getKey().startsWith(target)))
+          .collect(
+              Collectors.toMap(
+                  entry -> entry.getKey().replaceFirst("^" + target + ".", ""),
+                  Map.Entry::getValue));
+    }
+  }
+
+  private static class DeserializedRecord {
+    private final SinkRecord record;
+    private final boolean failed;
+
+    DeserializedRecord(SinkRecord record, boolean failed) {
+      this.record = record;
+      this.failed = failed;
+    }
+
+    public SinkRecord getRecord() {
+      return record;
+    }
+
+    public boolean isFailed() {
+      return failed;
+    }
+  }
+
+  private abstract static class ExceptionHandler {
```

Comment:
The plan is for this to be a configurable/injectable class.   Right now we just catch all exceptions. 

This won't work for all Converters and all transforms, particularly those that make network calls.   For example, AvroConverter throws exceptions around network failures (e.g. cannot contact schema registry) which should fail the SMT and not convert to a failed record. 

They are converter/SMT specific so should become a pluggable class along with some sensible defaults.